### PR TITLE
Slug transformer

### PR DESCRIPTION
**Pull request by ADmad**
Refs #321

### CODE REVIEW

**Comment by josegonzalez** [comment]
The class name is wrong. How are these tests passing?

**Comment by ADmad** [comment]
hah indeed, i wonder too.

**Comment by ADmad** [comment]
It does error out as expected if you run just `SlugTransformerTests.php`

**Comment by ADmad** [comment]
So phpunit was skipping the test file because i mistakenly named if `SlugTransformerTests.php` instead of `SlugTransformerTest.php`  (extra "s").

Getting the tests working also uncovered bugs :)

File: tests/TestCase/File/Transformer/SlugTransformerTest.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        $this->assertEquals(['path/to/file' => 'foo-e-a.txt'], $this->transformer->transform());
+    }
+
+    public function testTransformWithNoFileExt()
+    {
+        $entity = $this->getMock('Cake\ORM\Entity');
+        $table = $this->getMock('Cake\ORM\Table');
+        $data = ['tmp_name' => 'path/to/file', 'name' => 'foo é À'];
+        $transformer = new SlugTransformer($table, $entity, $data, 'field', []);
+        $this->assertEquals(['path/to/file' => 'foo-e-a'], $transformer->transform());
```
Comment (josegonzalez) [inline tests/TestCase/File/Transformer/SlugTransformerTest.php]: Shouldn't this have a `.txt` at the end?

**Comment by josegonzalez** [comment]
Not sure I understand why the tests suddenly don't skip, but w/e.

File: tests/TestCase/File/Transformer/SlugTransformerTest.php (hunk 1 above)
Comment (ADmad) [inline tests/TestCase/File/Transformer/SlugTransformerTest.php]: No, cause the name itself does not have an extension.

**Comment by ADmad** [comment]
> Not sure I understand why the tests suddenly don't skip, but w/e.

I mentioned it in previous comment. The test was skipped due to file name.

**Comment by ADmad** [comment]
@josegonzalez  Walking through the `beforeSave()` callback I realized that using this `InflectedTransfer` would be pointing. The file would be saved with inflected file name but the entity field would be still set to the original uninflected file name using `$path->filename()`.

So will have to scrap this and add an `InflectedProcessor` instead.

**Pull request merged**